src/solve.jl

```diff
@@ -391,7 +391,7 @@ function __init__()
 	end
 
 	@require NLopt="76087f3c-5699-56af-9a33-bf431cd00edd" begin
-		function __solve(prob::OptimizationProblem, opt::NLopt.Opt; maxiters::Number = 1000, nstart = 1, local_method = nothing, kwargs...)
+		function __solve(prob::OptimizationProblem, optAlg::Symbol; maxiters::Number = 1000, nstart = 1, local_method = nothing, kwargs...)
```

Comment:
I don't think we want to dispatch on symbol. That could lead to ambiguities down the line. It's best to make each dispatch have a clear type. Flux is the exception right now, and I want to fix that.

Reply:
That's what I suspected. Would it be OK to do: `opt::NLopt.Algorithm`? Then the user has to write, for instance, `solve(prob, NLopt.LD_MMA)`. I checked, it works.

Reply:
That could be fine, though we normally use instances and not types because then we can have default values for method-specific arguments. Can `NLopt.LD_MMA()` work?

Reply:
@ChrisRackauckas no that doesn't work and that was the reason we dispatched on `Opt` before. @mkg33 IMO this is not really needed and it actually should be covered in the docs just to clarify from our side. My reason for not liking the change is that we should try to stick where possible to the syntax from the optimiser package and not create confusion. I suspect the approach here would lead to more questions and justifications from our side down the line as compared to what we have where we can simply point out that we are only using what the optimisation package itself uses.

Reply:
I see what you mean, @Vaibhavdixit02. So I'll just provide an example and describe it in the documentation.